pkg/api/kubegraph/analysis/pod.go

```diff
@@ -27,7 +38,50 @@ func FindRestartingPods(g osgraph.Graph, f osgraph.Namer) []osgraph.Marker {
 		}
 
 		for _, containerStatus := range pod.Status.ContainerStatuses {
-			if containerStatus.RestartCount >= RestartThreshold {
+			switch {
+			case containerCrashLoopBackOff(containerStatus):
+				var suggestion string
+				switch {
+				case containerIsNonRoot(pod, containerStatus.Name):
+					suggestion = Df(`
+						The container is starting and exiting repeatedly. This usually means the container is unable
+						to start, misconfigured, or limited by security restrictions. Check the container logs with
+
+						  %s %s -c %s
+
+						Current security policy prevents your containers from being run as the root user. Some images
+						may fail expecting to be able to change ownership or permissions on directories. Your admin
+						can grant you access to run containers that need to run as the root user with this command:
+
+						  %s
+						`, logsCommandName, pod.Name, containerStatus.Name, fmt.Sprintf(securityPolicyCommandPattern, pod.Namespace, pod.Spec.ServiceAccountName))
+				default:
+					suggestion = Df(`
+						The container is starting and exiting repeatedly. This usually means the container is unable
+						to start, misconfigured, or limited by security restrictions. Check the container logs with
+
+						  %s %s -c %s
+						`, logsCommandName, pod.Name, containerStatus.Name)
+				}
+				markers = append(markers, osgraph.Marker{
+					Node: podNode,
+
+					Severity: osgraph.ErrorSeverity,
+					Key:      CrashLoopingPodError,
+					Message: fmt.Sprintf("container %q in %s is crash-looping", containerStatus.Name,
+						f.ResourceName(podNode)),
+					Suggestion: osgraph.Suggestion(suggestion),
+				})
+			case containerRestartedRecently(containerStatus, nowFn()):
+				markers = append(markers, osgraph.Marker{
+					Node: podNode,
+
+					Severity: osgraph.WarningSeverity,
+					Key:      RestartingPodWarning,
+					Message: fmt.Sprintf("container %q in %s has restarted within the last 10 minutes", containerStatus.Name,
```

Comment:
I considered it, but "10m" isn't readable enough.  I'll add a TODO that if we change it, someone has to change the message.